mscore/menus.cpp

```diff
@@ -878,7 +879,7 @@ PalettePanel* MuseScore::newNoteHeadsPalettePanel()
       QAction* action = s->action();
       QIcon icon(action->icon());
       ik->setAction("add-parentheses", icon);
-      sp->append(ik, s->help());
+      sp->append(ik, QT_TRANSLATE_NOOP("Palette", "Add parentheses to notehead"));
```

Comment:
Not sure this works here. Guess it needs to be `QApp->translate("Palette", "Add parentheses to notehead")`

Reply:
Do I have to do the same to the tempo text too (menus.cpp:1542 etc.)?

Reply:
Not sure

Reply:
I think a simple `tr("Add parentheses to notehead")` is good enough here because you are translating a fixed string. `qApp->translate()` is used to translate dynamic strings (i.e. a variable).

Reply:
`tr()` is likely to work incorrectly because of differences in translation contexts: strings read from `.workspace` files will be translated afterwards (in [`PaletteCell::translatedName()`](https://github.com/musescore/MuseScore/blob/master/mscore/palette/palettetree.cpp#L161)) using translation contexts defined in [`PaletteCell::translationContext()`](https://github.com/musescore/MuseScore/blob/master/mscore/palette/palettetree.cpp#L119) which wouldn't match the one used by `tr()` call (`Ms::MuseScore` in this case). Also in order to be compatible with those cell names which reuse strings from other contexts I had to make translation context for palette cell names dependent on element type. For `Icon`s the context named `action` is used so this string in `Palette` context will not actually be used.

Also for language changes to work properly it is indeed better to use `QT_TRANSLATE_NOOP` instead of setting an already translated string here, although this will be an issue only when launching MuseScore in non-English language and switching language after that.

By the way, "Add parentheses to notehead" string is already translatable and indeed reuses `help` string from the corresponding shortcut so there is no need to change `s->help()` to the string explicitly marked translatable in this particular case.

For other cases though my remark regarding translation contexts is valid.

Reply:
> By the way, "Add parentheses to notehead" string is already translatable and indeed reuses help string from the corresponding shortcut so there is no need to change s->help() to the string explicitly marked translatable in this particular case.

Actually I've changed that string from `s->help()` to "Add parentheses to notehead" because `s->help()` is actually "Add parentheses to element", and already exists in another place in the palette.

So should I change it to ` QT_TRANSLATE_NOOP("Palette", "Add parentheses to notehead")` or `` QT_TRANSLATE_NOOP("action", "Add parentheses to notehead")`` or something else?

Reply:
Ah, sorry, I didn't notice the difference.

As I mentioned above, the translation context used for this string should match the context used for translation of palette cells (which is defined by [this function](https://github.com/musescore/MuseScore/blob/master/mscore/palette/palettetree.cpp#L119)). As this particular element is of `Icon` type, either this string should be declared in `action` context (with `QT_TRANSLATE_NOOP("action", "Add parentheses to notehead")`), or `translationContext()` definition should be somehow changed to allow different contexts for these elements.

The same applies to any other string you declare here: translation context for it should match the one used to translate palette cell names, otherwise these strings will end up being not translated in UI.